### PR TITLE
Create n/w using azure verified module  avm-res-network-virtualnetwork instead of internal module

### PR DESCRIPTION
**Pull request by amitnamdeo**
Tested N/w AVM with both templates az-oci-adbs az-oci-exa-pdb successfully with avm n/w module usage

### CODE REVIEW

File: templates/az-oci-exa-pdb/main.tf [hunk 1]
```diff
@@ -3,33 +3,50 @@ resource "azurerm_resource_group" "resource_group" {
...
   providers = {
     azapi = azapi
   }
+  depends_on = [module.avm_vmc_network]
+
   exadata_infrastructure_resource_display_name                     = var.exadata_infrastructure_resource_display_name
   exadata_infrastructure_resource_name                             = var.exadata_infrastructure_resource_name
   location                                                         = var.location
-  oracle_database_delegated_subnet_id                              = module.vm_cluster_network.delegated_subnet_id
+  oracle_database_delegated_subnet_id                              = module.avm_vmc_network.subnets[0].id
```
Comment (acaifangyuan) [inline templates/az-oci-exa-pdb/main.tf]: subnets is a map
`module.avm_vmc_network.subnets["delegated"].id`

Comment (amitnamdeo) [reply templates/az-oci-exa-pdb/main.tf]: I tested `terraform plan` successfully with 
```
oracle_database_delegated_subnet_id = module.avm_vmc_network.subnets.delegated.resource_id
```

Comment (acaifangyuan) [reply templates/az-oci-exa-pdb/main.tf]: I ran apply and it failed due to module.avm_vmc_network.subnets[0].id

Comment (amitnamdeo) [reply templates/az-oci-exa-pdb/main.tf]: Can you please  pulled latest, after which you should see value as `module.avm_vmc_network.subnets.delegated.resource_id` not  `.subnets[0].id`

File: templates/az-oci-adbs/main.tf [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
-module "adbs_network" {
-  source = "../../modules/azure-vnet-subnet"
-  providers = {
-    azurerm = azurerm
+
+# Delete comment before merge
+# Reference code : https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/5b60a9a29fa471d3298427f706645e80f09de9da/examples/legacy_address_prefix/README.md 
+module "avm_odbas_network" {
+  source  = "Azure/avm-res-network-virtualnetwork/azurerm"
+  version = "0.2.4"
```
Comment (ziqi-li-oracle) [inline templates/az-oci-adbs/main.tf:12]: I wonder if the version is a required field here? Shall we remove the hardcoded value and let it use the latest one?

**Review by ziqi-li-oracle — APPROVED** [review]
LGTM

File: templates/az-oci-adbs/main.tf (hunk 2 above)
Comment (amitnamdeo) [reply templates/az-oci-adbs/main.tf:12]: yes as of now we can not use latest. We have landing zone's own tfm version restriction hence this has to stay

```
│ Error: Unsupported Terraform Core version
│ 
│   on .terraform/modules/avm-vnet-subnet/modules/subnet/terraform.tf line 2, in terraform:
│    2:   required_version = ">= 1.9.2"
```

**Review by acaifangyuan — APPROVED** [review]

**Pull request merged**